apps/nextjs/src/components/AppComponents/Chat/chat-panel.tsx

```diff
@@ -35,35 +35,48 @@ export function ChatPanel({ isDemoLocked }: Readonly<ChatPanelProps>) {
 
   const { trackEvent } = useAnalytics();
 
+  const sidebar = useSidebar();
+  const lessonPlanTracking = useLessonPlanTracking();
+
   const handleSubmit = useCallback(
     (value: string) => {
-      if (isLoading) return;
-      trackEvent("chat:send_message", {
-        id,
-        message: value,
-      });
-
-      append({
-        content: value,
-        role: "user",
-      }).catch((error) => {
-        Sentry.captureException(error);
-        toast.error("Failed to send message");
-      });
-    },
-    [isLoading, trackEvent, id, append],
-  );
+      setInput("");
+      if (sidebar.isSidebarOpen) {
+        sidebar.toggleSidebar();
+      }
 
-  const wrappedQueueUserAction = useCallback(
-    (action: Parameters<typeof queueUserAction>[0]) => {
-      queueUserAction(action).catch((error) => {
-        Sentry.captureException(error);
-        toast.error("Failed to queue action");
-      });
+      lessonPlanTracking.onSubmitText(value);
+
+      const isQueueable = ailaStreamingStatus === "Moderating";
+      if (isQueueable) {
+        queueUserAction(value);
+      } else {
+        trackEvent("chat:send_message", {
+          id,
+          message: value,
+        });
+
+        void append({
+          content: value,
+          role: "user",
+        });
```

Comment:
You can optionally await `append` to wait for the response. We don't need that so can discard the promise